pkgs/os-specific/linux/sssd/default.nix

```diff
@@ -39,7 +41,11 @@ stdenv.mkDerivation {
       --without-semanage
       --with-xml-catalog-path=''${SGML_CATALOG_FILES%%:*}
       --with-ldb-lib-dir=$out/modules/ldb
-      --with-nscd=${glibc.bin}/sbin/nscd
+      ${lib.optionalString (nscdConfig != null) ''
+        --with-nscd=${glibc.bin}/sbin/nscd
+        --with-nscd-conf=${nscdConfig}
```

Comment:
Ha, I was exactly asking myself what the _proper_ way to do this would be - so I let nscd create a symlink for its current configuration file in `/etc` and just point SSSD at its absolute path..?

I can elaborate on the SSSD-nscd specific bit. SSSD queries whether nscd is running (`--with-nscd`) and will additionally inspect its active configuration file (`--with-nscd-conf`). If it's not running, it won't do anything. If it is running and the configuration file doesn't disable passwd,  group, netgroup and services caching, it will print a warning to syslog because it essentially breaks SSSD. 

In case you're interested about the details, it's defined in `src/monitor.c`:
```c
    /* Warn if nscd seems to be running */
    ret = check_file(NSCD_SOCKET_PATH,
                     -1, -1, S_IFSOCK, S_IFMT, NULL, false);
    if (ret == EOK) {
        ret = sss_nscd_parse_conf(NSCD_CONF_PATH);

        switch (ret) {
            case ENOENT:
                sss_log(SSS_LOG_NOTICE,
                        "NSCD socket was detected. NSCD caching capabilities "
                        "may conflict with SSSD for users and groups. It is "
                        "recommended not to run NSCD in parallel with SSSD, "
                        "unless NSCD is configured not to cache the passwd, "
                        "group, netgroup and services nsswitch maps.");
                break;

            case EEXIST:
                sss_log(SSS_LOG_NOTICE,
                        "NSCD socket was detected and seems to be configured "
                        "to cache some of the databases controlled by "
                        "SSSD [fpasswd,group,netgroup,services]. It is "
                        "recommended not to run NSCD in parallel with SSSD, "
                        "unless NSCD is configured not to cache these.");
                break;

            case EOK:
                DEBUG(SSSDBG_TRACE_FUNC, "NSCD socket was detected and it "
                            "seems to be configured not to interfere with "
                            "SSSD's caching capabilities\n");
```